File: mockups/screenshot.py
from playwright.sync_api import sync_playwright
import os

html = os.path.abspath(os.path.join(os.path.dirname(__file__), 'article-wenkai.html'))
out  = os.path.abspath(os.path.join(os.path.dirname(__file__), 'article-wenkai-full.png'))

with sync_playwright() as p:
    browser = p.chromium.launch(
        executable_path='/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
    )
    page = browser.new_page(viewport={'width': 1080, 'height': 900})
    page.goto(f'file://{html}')
    page.wait_for_timeout(3000)  # 等字体+JS分页

    h = page.evaluate("document.getElementById('article').offsetHeight")
    page.set_viewport_size({'width': 1080, 'height': h})
    page.wait_for_timeout(500)

    page.locator('#article').screenshot(path=out, type='png')
    print(f'Done: 1080 x {h}px → {out}')
    browser.close()

File: mockups/screenshot.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
  const browser = await puppeteer.launch({ headless: true });
  const page = await browser.newPage();

  const filePath = path.resolve(__dirname, 'article-wenkai.html');
  await page.goto('file://' + filePath, { waitUntil: 'networkidle0' });

  // 等待字体和 JS 分页完成
  await new Promise(r => setTimeout(r, 2000));

  // 获取 article 元素的实际尺寸
  const dims = await page.evaluate(() => {
    const el = document.getElementById('article');
    return { width: el.offsetWidth, height: el.offsetHeight };
  });

  await page.setViewport({ width: dims.width, height: dims.height, deviceScaleFactor: 1 });

  // 截取 article 元素
  const el = await page.$('#article');
  await el.screenshot({
    path: path.resolve(__dirname, 'article-wenkai-full.png'),
    type: 'png',
  });

  console.log(`Done: ${dims.width} × ${dims.height}px`);
  await browser.close();
})();
